src/main/java/com/google/sps/data/EmailSender.java

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+// using SendGrid's Java Library
+// https://github.com/sendgrid/sendgrid-java
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Content;
+import com.sendgrid.helpers.mail.objects.Email;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import java.util.Map;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.URL;
+
+// Handles sending emails.
+@WebServlet("/email")
+public class EmailSender {
+  // The email messages are being sent from.
+  private final Email sender;
+  private final SendGrid sg;
+
+  public EmailSender(Email sender) throws Exception {
+    this.sender = sender;
+    this.sg =
+        new SendGrid(
+            new SecretFetcher("interview-me-step-2020").getSecretValue("SENDGRID_API_KEY"));
+  }
+
+  // Sends an email from the "from" Email to the "to" Email, with specified subject and content.
+  public Response sendEmail(Email recipient, String subject, Content content)
+      throws IOException, Exception {
+    Mail mail = new Mail(sender, subject, recipient, content);
+
+    Request request = new Request();
+    Response response;
+    request.setMethod(Method.POST);
+    request.setEndpoint("mail/send");
+    request.setBody(mail.build());
+    response = sg.api(request);
+    return response;
+  }
+
+  // Returns the contents of the file specified at filePath as a String. Useful for converting
+  // predifined email templates to text.
+  public static String fileContentToString(String filePath) throws IOException {
```

Comment:
No references to this function in this PR?

Reply:
https://github.com/googleinterns/InterviewMe/pull/115#discussion_r459443703
line 74 ^
lt's used in the test (not on Git) and will be used when we start to use email templates.